source/simulator/melt.cc

```diff
@@ -369,6 +380,12 @@ namespace aspect
 
       const double pressure_scaling = this->get_pressure_scaling();
 
+      std::vector<double> reactions(n_q_points);
+      const unsigned int porosity_index = introspection.compositional_index_for_name("porosity");
+      if (this->get_parameters().use_operator_splitting)
+        scratch.finite_element_values[introspection.extractors.compositional_fields[porosity_index]].get_function_values(this->get_reaction_vector(),
```

Comment:
what if ``use_operator_splitting`` is false, are we then using uninitialized ``reactions[q]`` below? Maybe initialize to NaN?